src/it/scala/nl/wbaa/gargoyle/proxy/S3ProxyItTest.scala

```diff
@@ -0,0 +1,188 @@
+    *
+    * @param size     Size in Bytes of the file you want to test
+    * @param testCode Code that accepts the name of the created file
+    * @return Assertion
+    */
+  def withFile(size: Long)(testCode: String => Assertion): Assertion = {
+    val filename = "testfile.tst"
+    val file = new RandomAccessFile(filename, "rw")
+    file.setLength(size)
+    file.close()
+
+    try {
+      testCode(filename)
+    } finally {
+      new File(filename).delete()
+    }
+  }
+
+  // TODO: Expand with different signer types
+  val awsSignerTypes = List(
+    "S3SignerType" //,
+    //      SignerFactory.NO_OP_SIGNER,
+    //      SignerFactory.QUERY_STRING_SIGNER,
+    //      SignerFactory.VERSION_FOUR_SIGNER,
+    //      SignerFactory.VERSION_FOUR_UNSIGNED_PAYLOAD_SIGNER,
+    //      SignerFactory.VERSION_THREE_SIGNER
+  )
+
+  awsSignerTypes.foreach { awsSignerType =>
+
+    "S3 Proxy" should {
+      s"proxy with $awsSignerType" that {
+
+        "list the current buckets" in withSdkToTestProxy(awsSignerType) { sdk =>
+          assert(sdk.listBuckets().asScala.toList.map(_.getName) == List("demobucket"))
+        }
+
+        "create and remove a bucket" in withSdkToTestProxy(awsSignerType) { sdk =>
+          sdk.createBucket("createbuckettest")
+          assert(sdk.listBuckets().asScala.toList.map(_.getName).contains("createbuckettest"))
+          sdk.deleteBucket("createbuckettest")
+          assert(!sdk.listBuckets().asScala.toList.map(_.getName).contains("createbuckettest"))
+        }
+
+        "list files in a bucket" in withSdkToTestProxy(awsSignerType) { sdk =>
+          sdk.putObject("demobucket", "keyListFiles", "content")
+          val resultV2 = sdk.listObjectsV2("demobucket").getObjectSummaries.asScala.toList.map(_.getKey)
+          val result = sdk.listObjects("demobucket").getObjectSummaries.asScala.toList.map(_.getKey)
+
+          assert(resultV2.contains("keyListFiles"))
+          assert(result.contains("keyListFiles"))
+        }
+
+        "check if bucket exists" in withSdkToTestProxy(awsSignerType) { sdk =>
+          assert(sdk.doesBucketExistV2("demobucket"))
+        }
+
+        "put, get and delete an object from a bucket" in withSdkToTestProxy(awsSignerType) { sdk =>
+          withFile(1024 * 1024) { filename =>
+            // PUT
+            sdk.putObject("demobucket", "keyPutFileByContent", "content")
+            sdk.putObject("demobucket", "keyPutFileByFile", new File(filename))
+
+            // GET
+            val checkContent = sdk.getObjectAsString("demobucket", "keyPutFileByContent")
+            assert(checkContent == "content")
+            val keys1 = getKeysInBucket(sdk)
+            List("keyPutFileByContent", "keyPutFileByFile").map(k => assert(keys1.contains(k)))
+
+            // DELETE
+            sdk.deleteObject("demobucket", "keyPutFileByContent")
+            val keys2 = getKeysInBucket(sdk)
+            assert(!keys2.contains("keyPutFileByContent"))
```

Comment:
a lot duplicated strings (keyPutFileByContent, demobucket, ....) - could be mistyped. Better to declare vals and then IDE helps

Reply:
done